core/src/test/java/org/apache/iceberg/TestSingleValueParser.java

```diff
@@ -157,6 +167,15 @@ public void testInvalidTimestamptz() {
         .hasMessageStartingWith("Cannot parse default as a timestamptz value");
   }
 
+  @Test
+  public void testInvalidGeometry() {
+    Type expectedType = Types.GeometryType.get();
+    String defaultJson = "\"POINT (1 2 3 4 5 6)\"";
+    assertThatThrownBy(() -> defaultValueParseAndUnParseRoundTrip(expectedType, defaultJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageMatching("Cannot parse default as a geometry.* value.*");
```

Comment:
What are the `.*` patterns matching?

Reply:
Geometry types without CRS were serialized as `geometry()`. The last commit fixed this so `.*` pattern was removed.